server/server.go

```diff
@@ -0,0 +1,67 @@
+package server
+
+import (
+	"io"
+	"log"
+	"net/http"
+	"strconv"
+
+	"github.com/gorilla/mux"
+)
+
+type ServerOptions struct {
+	Token     string
+	BaseURL   string
+	PublicURL string
+	Bind      string
+	Port      int
+}
+
+type Server struct {
+	status ServerStatus
+}
+
+func (self *Server) Start(options *ServerOptions) error {
+	self.reload()
+
+	router := self.createRouter()
+	return listenAndServe(options.Bind, options.Port, router)
+}
+
+func (self *Server) reload() {
+	self.status = StatusMaintenance
+	// reload project information
+	// ensure each project has webhook set
+	self.status = StatusOK
+}
+
+func (self *Server) createRouter() http.Handler {
+	router := mux.NewRouter()
+
+	router.HandleFunc("/reload", reloadHandler(self))
+
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
```

Comment:
Do you really need a "middleware" for that? Is not possible solve this inside the `reloadHandler`?